Sources/SwiftSDKGenerator/SystemUtils/Shell.swift

```diff
@@ -130,61 +106,41 @@ final class Shell {
 
   static func readStdout(
     _ command: String,
-    currentDirectory: FilePath? = nil,
     shouldLogCommands: Bool = false,
     file: String = #file,
     line: Int = #line
   ) async throws -> String {
-    let process = try Shell(
-      command,
-      currentDirectory: currentDirectory,
-      shouldDisableIOStreams: false,
-      shouldLogCommands: shouldLogCommands,
-      file: file,
-      line: line
+    if shouldLogCommands {
+      print(command)
+    }
+
+    let result = try await ProcessExecutor.runCollectingOutput(
+      executable: "/bin/sh",
+      ["-c", command],
+      collectStandardOutput: true,
+      collectStandardError: false,
+      perStreamCollectionLimitBytes: 100 * 1024 * 1024,
+      environment: ProcessInfo.processInfo.environment
     )
 
-    try await process.waitUntilExit()
+    try result.exitReason.throwIfNonZero()
 
-    var output = ""
-    for try await chunk in process.stdout {
-      output.append(String(data: chunk, encoding: .utf8)!)
+    guard
+      var buffer = result.standardOutput,
+      let result = buffer.readString(length: buffer.readableBytes)
+    else {
+      throw GeneratorError.noProcessOutput(command)
     }
-    return output
+
+    return result
   }
 }
 
-@available(*, unavailable)
-extension Shell: Sendable {}
-
-private extension AsyncThrowingStream where Element == Data, Failure == any Error {
-  init(
-    _ process: Process,
-    pipeKeyPath: ReferenceWritableKeyPath<Process, Any?>,
-    commandInfo: CommandInfo
-  ) {
-    self.init { continuation in
-      let pipe = Pipe()
-      pipe.fileHandleForReading.readabilityHandler = { [unowned pipe] fileHandle in
-        let data = fileHandle.availableData
-        if !data.isEmpty {
-          continuation.yield(data)
-        } else {
-          if !process.isRunning && process.terminationStatus != 0 {
-            continuation.finish(
-              throwing: FileOperationError.nonZeroExitCode(process.terminationStatus, commandInfo)
-            )
-          } else {
-            continuation.finish()
-          }
-
-          // Clean up the handler to prevent repeated calls and continuation finishes for the same
-          // process.
-          pipe.fileHandleForReading.readabilityHandler = nil
-        }
-      }
-
-      process[keyPath: pipeKeyPath] = pipe
+extension ChunkSequence {
+  func printChunksAsStrings() async throws {
+    for try await var chunk in self {
+      guard let string = chunk.readString(length: chunk.readableBytes) else { continue }
+      print(string)
```

Comment:
Couple of things:
- no need to `var chunk`, please `let chunk`
- instead of `guard let string = chunk.readString(length: chunk.readableBytes) else { continue }` please use `String(buffer: chunk)`, easier and better
- if you want to print line by line it'd be

```
for try await let line in self.splitIntoLines() {
    print(line)
}
```

Reply:
That should correct the output and is also much shorter.

Reply:
Alternatively, there are a bunch of helpers such as

```
        try await ProcessExecutor.runProcessingOutput(
            executable: "/bin/foo",
            [ "bar", "buz" ],
            outputProcessor: { stream, line in
                print(line, "(from \(stream))")
            },
            splitOutputIntoLines: true,
        ).throwIfNonZero()
```

the `outputProcessor` allows you to specify a quick closure if your processing is very limited

Reply:
Or of course

```swift
            let exe = ProcessExecutor(
                executable: "/usr/bin/foo", ["bar"],
            )
            async let result = exe.run()
            for try await line in await merge(exe.standardOutput.splitIntoLines(dropTerminator: true).strings, exe.standardError.splitIntoLine(dropTerminator: true).strings) {
                print(line)
            }
            try await result.throwIfNonZero() // throw if process exited non-zero
```

if you don't want to use the `runProcessingOutput` convenience